src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,299 @@
+        '__' + query_template_filename.replace('.', '-')
+    if cache_suffix:
+        results_dirname = results_dirname + '__' + cache_suffix
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    results_filename = 'results.csv'
+    command_save_filename = 'command.sh'
+    results_path = os.path.join(results_dirpath, results_filename)
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    instantiated_query_path = os.path.join(results_dirpath, 'query.sparql')
+    command_str = f'robot query --input {onto_path} --query {instantiated_query_path} {results_path}'
+
+    # Instantiate template
+    with open(query_template_path, 'r') as f:
+        template_str = f.read()
+    template_obj = Template(template_str)
+    if not terms and not regexp:
+        raise RuntimeError('sparql_jinja2_robot_query: Must pass either `terms` or `regexp`.')
+    elif terms and regexp:
+        raise RuntimeError('sparql_jinja2_robot_query: Must pass only 1: `terms` or `regexp`.')
+    elif terms:
+        instantiated_str = template_obj.render(
+            property_path_qualifier='*' if include_superclass else '+',
+            values=' '.join(terms))
+    elif regexp:
+        instantiated_str = template_obj.render(regexp=regexp)
+
+    # Cache and run
+    os.makedirs(results_dirpath, exist_ok=True)
+    if not (os.path.exists(results_path) and use_cache):
+        with open(instantiated_query_path, 'w') as f:
+            # noinspection PyUnboundLocalVariable
+            f.write(instantiated_str)
+        with open(command_save_path, 'w') as f:
+            f.write(command_str)
+        subprocess.run(command_str.split())
+
+    # Read results and return
+    try:
+        df = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError:
+        # remove: so that it doesn't read this from cache, though it could be that there were really no results.
+        os.remove(results_path)
+        # could also do `except pd.errors.EmptyDataError as err`, `raise err` or give option for this as a param to func
+        df = pd.DataFrame()  # could also return None
+
+    return df
+
+
+def expand_ontological_exclusions(
+    onto_path: str, exclusions_df: pd.DataFrame, cache_suffix: str = None,
+    export_fields=['term_id', 'exclusion_reason'], uri_fields=['term_id', 'child_id']
+) -> pd.DataFrame:
+    """Using an ontology file and an exclusions file, query and get return an extensional list of exclusions.
+    cache_suffix: Used by and explained in sparql_jinja2_robot_query()."""
+    # Variable names: kids0=Children excluded; kids1=Children included
+    df_kids1 = exclusions_df[exclusions_df['exclude_children'] == False]
+    df_kids0 = exclusions_df[exclusions_df['exclude_children'] == True]
+    terms_kids1: List[str] = list(set(df_kids1['term_id']))  # QA: set() removes any duplicates
+    df_kids1_results: pd.DataFrame = sparql_jinja2_robot_query(
```

Comment:
FYI: For cases where `exclude_children=FALSE`, we need to query. But when `TRUE`, don't need to query. So I split rows, query for the ones where `FALSE`, and then I join the results together at the end.

Reply:
resolving for fyi reduction as requested